### PR TITLE
adding type annotations and png example script

### PR DESCRIPTION
**Pull request by FoamyGuy**
Follow up on #61 to add typing annotations and an example of the new PNG loading functionality.

### CODE REVIEW

**Comment by tekktrik** [comment]
Converted to draft to prevent merging, since this PR will help test new CI changes.

File: adafruit_imageload/png.py [hunk 1]
```diff
@@ -13,16 +13,22 @@
...
 
-
 __version__ = "0.0.0-auto.0"
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_ImageLoad.git"
 
 
 def load(
-    file, *, bitmap, palette=None
+    file: str, *, bitmap: BitmapConstructor, palette: PaletteConstructor = None
 ):  # pylint: disable=too-many-locals,too-many-branches
```
Comment (deshipu) [inline adafruit_imageload/png.py:32]: I have been researching a little about how to annotate the return type of this correctly, without assuming that we are always going to use the `displayio.Bitmap` and `displayio.Palette`. It seems like we can do this by using type aliases (https://peps.python.org/pep-0484/#type-aliases) to make sure the function returns the same type as the constructor that it gets, and optionally maybe protocols (https://peps.python.org/pep-0544/) to make sure those types have the required methods on them.

File: examples/imageload_png_simpletest.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import adafruit_imageload
+
+# built-in display
+display = board.DISPLAY
+
+# load png image
+image, palette = adafruit_imageload.load("images/test_image.png")
+
+# Set the transparency index color to be hidden
+palette.make_transparent(0)
```
Comment (deshipu) [inline examples/imageload_png_simpletest.py:16]: This is another todo – we could read the transparency information from the file, there are two kinds of chunks that can define that.

**Review by deshipu — APPROVED** [review]

**Comment by FoamyGuy** [comment]
I've opened #65 try track transparency implementation issue. #62 also is there for the typing improvements with protocols.

Merging this one based on the approval to get it cleared out and get the example published for folks to find and use.

**Pull request merged**